### PR TITLE
feat: task use existing run file

### PR DESCRIPTION
**Pull request by yukun-dong**
https://github.com/microsoft/vscode-dapr/issues/272

### CODE REVIEW

File: src/commands/scaffoldDaprTasks.ts [hunk 1]
```diff
@@ -134,6 +138,89 @@ async function createUniqueName(prefix: string, isUnique: ConflictUniquenessPred
...
+            return { 'type': 'overwrite' };
+        } else {
+            name = await createUniqueName(localize('commands.scaffoldDaprTasks.configurationPrefix', '{0} - ', name), isUnique);
+
+            return { 'type': 'rename', name };
+        }
+    };
+
+    if (fs.existsSync(runFilePath)) {
+        const runFileTask: vscode.MessageItem = { title: 'yes' };
```
Comment (philliphoff) [inline src/commands/scaffoldDaprTasks.ts:190]: Nit: these strings should be localized.

File: src/commands/scaffoldDaprTasks.ts [hunk 2]
```diff
@@ -134,6 +138,89 @@ async function createUniqueName(prefix: string, isUnique: ConflictUniquenessPred
...
+        const defaultTask: vscode.MessageItem = { title: 'no' };
+        const result = await ui.showWarningMessage('You already have a Dapr run file. Would you like to use it in the scaffolded task?',
+            { modal: true },
+            runFileTask, defaultTask);
+
+        if (result === runFileTask) {
```
[the remark this answers is not in the document]
Comment (yukun-dong) [reply src/commands/scaffoldDaprTasks.ts:197]: In my opinion it's ok to have 2 sets of defining tasks and configurations. Below are the reasons:

1. Although they are all dapr related tasks and configurations, the properties in them are quite different. 
2. The flow to generate them is different: dapr run file doesn't require additional user input, not using dapr run file requires user input of appId, appPort, etc.

Therefore, I don't think it's convenient to define these task and configuration in one place.

Please let me know if you have additional comments on this. Thank you!

[the remark this answers is not in the document]
Comment (philliphoff) [reply src/commands/scaffoldDaprTasks.ts:197]: In that case, then we should use constants that are shared between them as much as possible, to ensure that values stay the same between versions.

File: src/commands/scaffoldDaprTasks.ts [hunk 3]
```diff
@@ -242,73 +167,199 @@ export async function scaffoldDaprTasks(context: IActionContext, scaffolder: Sca
...
+        if (result === overwrite) {
+            return { 'type': 'overwrite' };
+        } else {
+            name = await createUniqueName(localize('commands.scaffoldDaprTasks.configurationPrefix', '{0} - ', name), isUnique);
+
+            return { 'type': 'rename', name };
+        }
+    };
+
+    if (fs.existsSync(runFilePath)) {
```
Comment (philliphoff) [inline src/commands/scaffoldDaprTasks.ts:190]: Nit: when using `fs` we should use the async versions (in `fs/promises`) when possible. That said, we should try to use VS Code's [filesystem API](https://code.visualstudio.com/api/references/vscode-api#FileSystem) even before that, as that helps enable the extension to run in virtualized environments (like browsers). (There are still some references to `fs/promises` in the extension that we should replace when convenient, but ideally we should not create new ones.)

File: src/commands/scaffoldDaprTasks.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yukun-dong) [reply src/commands/scaffoldDaprTasks.ts:201]: The 2 sets of task and configuration are listed below. The only shared constants are type: "dapr" and others are quite different. So I used constants for "dapr" and others remain the same. I wonder if that's ok.

Not using runFile:
```json
        {
            "appId": "app",
            "appPort": 5074,
            "label": "dapr-debug",
            "type": "dapr",
            "dependsOn": "build"
        },
        {
            "appId": "app",
            "label": "daprd-down",
            "type": "daprd-down"
        },
```
```json
        {
            "name": ".NET Core Launch (web) with Dapr",
            "type": "coreclr",
            "request": "launch",
            "preLaunchTask": "dapr-debug",
            "program": "${workspaceFolder}/bin/Debug/net7.0/netcoreapp.dll",
            "args": [],
            "cwd": "${workspaceFolder}",
            "stopAtEntry": false,
            "serverReadyAction": {
                "action": "openExternally",
                "pattern": "\\bNow listening on:\\s+(https?://\\S+)"
            },
            "env": {
                "ASPNETCORE_ENVIRONMENT": "Development"
            },
            "sourceFileMap": {
                "/Views": "${workspaceFolder}/Views"
            },
            "postDebugTask": "daprd-down"
        },
```
Using runFile:
```json
        {
            "label": "dapr",
            "type": "dapr",
            "runFile": "$(workspaceFolder)/dapr.yaml"
        }
```
```json
        {
            "name": "Launch Dapr",
            "request": "launch",
            "type": "dapr",
            "runFile": "$(workspaceFolder)/dapr.yaml",
            "preLaunchTask": "dapr"
        }
```

File: src/commands/scaffoldDaprTasks.ts (hunk 3 above)
Comment (yukun-dong) [reply src/commands/scaffoldDaprTasks.ts]: Thanks! I updated it to use `vscode.workspace.fs.stat()` method.

File: src/commands/scaffoldDaprTasks.ts [hunk 4]
```diff
@@ -242,73 +171,199 @@ export async function scaffoldDaprTasks(context: IActionContext, scaffolder: Sca
...
+            return { 'type': 'overwrite' };
+        } else {
+            name = await createUniqueName(localize('commands.scaffoldDaprTasks.configurationPrefix', '{0} - ', name), isUnique);
+
+            return { 'type': 'rename', name };
+        }
+    };
+
+    if (await vscode.workspace.fs.stat(vscode.Uri.file(runFilePath)).then(() => true, () => false)) {
+        const runFileTask: vscode.MessageItem = { title: localize('commands.scaffoldDaprTasks.useExistingDaprRunFile', 'yes') };
```
Comment (philliphoff) [inline src/commands/scaffoldDaprTasks.ts:195]: Rather than `yes` and `no`, I think I'd go with something like `Use Run File` and `Run Dapr Directly`.

**Review by philliphoff — APPROVED** [review]
Just one comment re: text in user prompts but otherwise looks ready to go.

**Pull request merged**